### PR TITLE
add support for Ubuntu 20.04 (PSRE-203)

### PR DESCRIPTION
**Pull request by nadeemshahzad**
Configuration Pull Request
---

Make sure that the following steps are done before merging:

  - [ ] A DevOps team member has approved the PR if it is code shared across multiple services and you don't own all of the services.
  - [ ] Are you adding any new default values that need to be overridden when this change goes live? If so:
    - [ ] Update the appropriate internal repo (be sure to update for all our environments)
    - [ ] If you are updating a secure value rather than an internal one, file a DEVOPS ticket with details.
    - [ ] Add an entry to the CHANGELOG.
  - [ ] If you are making a complicated change, have you performed the proper testing specified on the [Ops Ansible Testing Checklist](https://openedx.atlassian.net/wiki/display/EdxOps/Ops+Ansible+Testing+Checklist)?  Adding a new variable does not require the full list (although testing on a sandbox is a great idea to ensure it links with your downstream code changes).
  - [ ] Think about how this change will affect Open edX operators.  Have you updated the wiki page for the next Open edX release?

### CODE REVIEW

File: playbooks/roles/common/tasks/main.yml [hunk 1]
```diff
@@ -106,6 +104,19 @@
     update_cache: yes
   when: ansible_distribution in common_debian_variants
 
+# Pip for Python2.7 is not included in the Ubuntu 20.04 repositories
```
Comment (fredsmith) [inline playbooks/roles/common/tasks/main.yml]: why are we installing python 2.7?

Comment (nadeemshahzad) [reply playbooks/roles/common/tasks/main.yml]: @fredsmith Installing python 2.7 as Ansible tries to use python 2 by default and can not even make it past the Gathering facts task so we will not able to ```use ansible_python_interpreter: /usr/bin/python3``` in our configuration for Ubuntu 20.04. Ansible 2.8 was shipped with interpreter discovery (https://github.com/ansible/ansible/blob/stable-2.8/changelogs/CHANGELOG-v2.8.rst#major-changes ), we can update the ansible in configuration with 2.8 release and can completely get rid of python 2  on Ubuntu 20.04.

File: playbooks/roles/mysql/tasks/mysql.yml [hunk 2]
```diff
@@ -33,19 +33,62 @@
...
+  apt_repository:
+    repo: "{{ MYSQL_REPO }}"
+    state: present
+  when: ansible_distribution_release == 'focal'
+
+# Installing mysql-5.7-client, mysql-5.7-community-server
+# and mysql-5.7-server in separate tasks to resolve dependencies
+- name: install mysql-5.7-client
+  apt:
+    name: mysql-client=5.7.31-1ubuntu18.04
```
Comment (adzuci) [inline playbooks/roles/mysql/tasks/mysql.yml]: It looks like you define `{{ mysql_pkgs }}` for mysql 5.6 in defaults, but hard code 5.7 here, any reason for that? https://github.com/edx/configuration/pull/5974/files#diff-64d565c7641d1d6d0f05105b2ab64fe7R44

File: playbooks/roles/nginx/defaults/main.yml [hunk 3]
```diff
@@ -128,7 +128,7 @@ nginx_user: root
 nginx_htpasswd_file: "{{ nginx_app_dir }}/nginx.htpasswd"
 nginx_default_sites: []
 nginx_debian_pkgs:
-  - python-passlib
+    - python3-passlib
```
Comment (adzuci) [inline playbooks/roles/nginx/defaults/main.yml]: I don't think you needed to add two spaces in front of this.

File: playbooks/roles/notifier/tasks/deploy.yml [hunk 4]
```diff
@@ -49,6 +49,7 @@
   pip:
     requirements: "{{ NOTIFIER_REQUIREMENTS_FILE }}"
     virtualenv: "{{ NOTIFIER_VENV_DIR }}"
+    virtualenv_python: "{{ notifier_python_version }}"
```
Comment (adzuci) [inline playbooks/roles/notifier/tasks/deploy.yml]: Can we just get rid of notifier?

**Review by adzuci — APPROVED** [review]
I left a few comments, but I think this should be ok to merge from our perspective.  I am curious if OpenCraft may be using the bootstrap script for instances that need Python 2.7 to be explicitly installed.

cc @bradenmacdonald do you feel that this change will impact you guys and would it make sense to get a review from someone on your team before we merge this change into master?

File: playbooks/roles/mysql/tasks/mysql.yml (hunk 2 above)
Comment (nadeemshahzad) [reply playbooks/roles/mysql/tasks/mysql.yml]: Updated to use vars for mysql 5.7 pkgs

File: playbooks/roles/nginx/defaults/main.yml (hunk 3 above)
Comment (nadeemshahzad) [reply playbooks/roles/nginx/defaults/main.yml]: fixed

File: playbooks/roles/notifier/tasks/deploy.yml (hunk 4 above)
Comment (nadeemshahzad) [reply playbooks/roles/notifier/tasks/deploy.yml]: Removed notifier from sandbox builds

**Comment by nadeemshahzad** [comment]
Ansible 2.5 and above work with python 3 and from release 2.8 it can correctly discover the default python interpreter on remote hosts ( https://github.com/ansible/ansible/blob/stable-2.8/changelogs/CHANGELOG-v2.8.rst#major-changes ). We have upgraded the ansible to 2.8.18 (https://github.com/edx/configuration/pull/5986/files) and now we can safely remove python2.7 packages for Ubuntu xenial and above.  I have successfully build the sandboxes for Ubuntu xenial and Bionic. 
https://tools-edx-jenkins.edx.org/job/Sandboxes/job/CreateSandbox/32757/
https://tools-edx-jenkins.edx.org/job/Sandboxes/job/CreateSandbox/32759/

**Comment by symbolist** [comment]
@adzuci @nadeemshahzad Thanks for the heads up! @bradenmacdonald asked me to take a quick look here. As long as support for Ubuntu 18.04 is not being dropped, it should be perfectly fine (we can resolve any small issues that come up quickly). From a scan it does not look like this PR does drop that?

**Pull request merged**